app/src/androidTest/java/com/stevesoltys/seedvault/e2e/BackupRestoreTest.kt

```diff
@@ -0,0 +1,139 @@
+package com.stevesoltys.seedvault.e2e
+
+import androidx.test.filters.LargeTest
+import com.stevesoltys.seedvault.e2e.screen.impl.RestoreScreen
+import com.stevesoltys.seedvault.restore.RestoreViewModel
+import com.stevesoltys.seedvault.transport.backup.PackageService
+import com.stevesoltys.seedvault.ui.notification.BackupNotificationManager
+import io.mockk.every
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.junit.Test
+import org.koin.core.component.inject
+import java.util.concurrent.atomic.AtomicBoolean
+
+@LargeTest
+class BackupRestoreTest : LargeTestBase() {
+
+    private val packageService: PackageService by inject()
+
+    private val spyBackupNotificationManager: BackupNotificationManager by inject()
+
+    private val restoreViewModel: RestoreViewModel by inject()
+
+    companion object {
+        private const val BACKUP_TIMEOUT = 360 * 1000L
+        private const val RESTORE_TIMEOUT = 360 * 1000L
+    }
+
+    @Test
+    fun `backup and restore applications`() = run {
+        launchBackupActivity()
+        verifyCode()
+        chooseBackupLocation()
+
+        val eligiblePackages = launchAllEligibleApps()
+        performBackup(eligiblePackages)
+        uninstallPackages(eligiblePackages)
+        performRestore()
+
+        val packagesAfterRestore = getEligibleApps()
+        assert(eligiblePackages == packagesAfterRestore)
```

Comment:
What apps are installed on the emulator for testing? Do those apps even have data to get backed up? Do we have a way to assert that the data in the backup really matches what we would expect to get restored?

Reply:
Related ticket: https://github.com/seedvault-app/seedvault/issues/548

Reply:
Created a repository for storing these test backups here:

https://github.com/seedvault-app/seedvault-test-data/releases/tag/1

I will work on another PR to start using these

Reply:
We are now using the above as a baseline for the E2E test.

The tests will:
1. Provision the emulator with Seedvault.
2. Download and restore the test backup.
3. Run the test afterwards, with a nice set of data and apps.

It will now also record the entire process and upload the recording as an artifact!

Reply:
@grote as of the latest commit we are now also:

1. Intercepting the backup and restore data, hashing and storing it for each package for both K/V and Full.
2. Asserting that the data we intercepted during restore matches the backup data.
3. Asserting that the backup metadata has no `UNKNOWN_ERROR` cases.

Reply:
awesome! 

maybe later, we can also check that certain packages in the test data did get backed up (maybe assert the number of packages as well?) and maybe, if test data is fixed, for some packages compare against expected data hash?

Reply:
Yes definitely! All of those would be great improvements. Test data is static as you said so should work.